### ISSUE TITLE
[Feature Request] Adaptive Gradient Clipping

### ISSUE THREAD

**Issue by kayuksel** [jettify/pytorch-optimizer#272]
It would be cool to use AGC from the following paper with any optimizer in this package: https://arxiv.org/abs/2102.06171

**Comment by kayuksel** [comment]
Here is an implementation for Adaptive Gradient Clipping: https://github.com/vballoli/nfnets-pytorch/blob/main/nfnets/agc.py

**Comment by jettify** [comment]
Would you like to create PR?
